Add management methods

mroz22 opened the pull request:
opening discussion

- [ ] shouldt management methods have useEmptyPassphrase = true as default?
- [ ] camelCase/snake_case disorder. Connect does not have 100% clear convention which case should be used for calls input params. For example here https://github.com/trezor/connect/blob/65b4a064125e5e65fb5010b0fd0b43a75d06238d/src/js/core/methods/CardanoSignTransaction.js#L36 and here https://github.com/trezor/connect/blob/65b4a064125e5e65fb5010b0fd0b43a75d06238d/src/js/core/methods/ResetDevice.js#L31 

- [ ] naming - stick to protobuff definitions or make own names where it make sense? Just one case comes to my mind now - protobuff has `recoveryDevice`, nicer would be `recoverDevice`

szymonlesisz commented on src/js/core/methods/helpers/paramsValidator.js:
```diff
@@ -6,9 +6,10 @@ import { fromHardened } from '../../../utils/pathUtils';
 type Param = {
     name: string,
-    type?: string,
+    type?: 'string' | 'number' | 'array' | 'array buffer' | 'boolean' | 'amount' | 'object',
```
make "arraybuffer" without space

mroz22 replied:
outdated, I have switched to plain buffer.

szymonlesisz commented on src/js/core/Core.js:
```diff
@@ -425,7 +425,17 @@ export const onCall = async (message: CoreMessage): Promise<void> => {
+                typeof UI.FIRMWARE_NOT_SUPPORTED
+            ) = device.hasUnexpectedMode(method.requiredFirmware, method.allowDeviceMode);
+
+            console.warn('unexpectedMode', unexpectedMode);
```
remove console

mroz22 committed:
add management methods

update features after all device calls

add recoveryDevice method

fix missing props

add unknown firmware status

firmware status none, hacking reloadign features (to discuss)

skipFinalReload for firmwareErase, change buffer validation

back to array-buffer

wip

wip

fix

szymonlesisz commented on Makefile:
```diff
@@ -27,8 +27,8 @@ build-connect:
-	rsync -avz --delete -e ssh ./build/* admin@dev.sldev.cz:~/sisyfos/www/connect
+	sed -i '' -e 's/connect.trezor.io/sisyfos.trezor.io\/martin-connect/g' ./build/js/iframe.*.js
+	rsync -avz --delete -e ssh ./build/* admin@dev.sldev.cz:~/sisyfos/www/martin-connect
 
```
remove custom build, just drop changes in this file

mroz22 committed:
fix fw update

mroz22 committed:
support for advanced recovery

szymonlesisz commented on src/js/core/methods/ApplySettings.js:
```diff
@@ -0,0 +1,41 @@
+        ]);
+
+        this.params = {
+            label: payload.label,
```
why validating multiple fields when you are using only label? Or is it not finished?

szymonlesisz commented on src/js/core/methods/helpers/uploadFirmware.js:
```diff
@@ -0,0 +1,36 @@
+): trezor.Success => {
+    let response: MessageResponse<trezor.Success | FirmwareRequest$> = {};
+
+    if (model === 1) {
```
i don't see this logic with strip 256 bytes for BL > 1.80? Shouldn't it be here?

szymonlesisz committed:
[WIP] ApplySettings with confirmation

szymonlesisz committed:
allow management method only for origins from config

mroz22 committed:
add popup for management methods

mroz22 committed:
Review fixes

mroz22 committed:
revert accidental config change

szymonlesisz committed:
update messages.json

szymonlesisz committed:
flowtype: add skipFinalReload to CommonParams

szymonlesisz committed:
fix popup word input

szymonlesisz committed:
flowtype add display_rotation to ApplySettings

szymonlesisz commented:
good job!
